libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -360,87 +323,86 @@ bool PbftManager::resetRound_() {
   }
 
   LOG(log_nf_) << "Determined round from votes: " << consensus_pbft_round;
-  round_clock_initial_datetime_ = now_;
+  shared_state_->round_clock_initial_datetime_ = shared_state_->now_;
   // Update current round and reset step to 1
   round_ = consensus_pbft_round;
   resetStep_();
-  state_ = value_proposal_state;
+  shared_state_->state_ = propose;
+  current_step_ = std::make_unique<step::Propose>(shared_state_, node_);
```

Comment:
I would create some kind of factory function that will just take `step::Propose` type as arg and will do this thing. But this is just my suggestion, so you do not need to do `shared_state_, node_` all the time

Reply:
I just decided to left it for next task(my current task), because it will be changed in it even more I think